Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp

```diff
@@ -69,60 +70,82 @@ WebFoundTextRangeController::WebFoundTextRangeController(WebPage& webPage)
 {
 }
 
-void WebFoundTextRangeController::findTextRangesForStringMatches(const String& string, OptionSet<FindOptions> options, uint32_t maxMatchCount, CompletionHandler<void(HashMap<WebCore::FrameIdentifier, Vector<WebFoundTextRange>>&&)>&& completionHandler)
+static inline WebFoundTextRange createWebFoundTextRange(SimpleRange& simpleRange)
 {
-    auto result = protectedWebPage()->protectedCorePage()->findTextMatches(string, core(options), maxMatchCount, false);
-    Vector<WebCore::SimpleRange> findMatches = WTFMove(result.ranges);
+    Ref document = simpleRange.startContainer().document();
+    RefPtr element = document->documentElement();
+    RefPtr frame = document->frame();
 
-    if (!findMatches.isEmpty())
-        m_cachedFoundRanges.clear();
+    // FIXME: We should get the character ranges at the same time as the SimpleRanges to avoid additional traversals.
+    CharacterRange range = characterRange(makeBoundaryPointBeforeNodeContents(*element), simpleRange, WebCore::findIteratorOptions());
+    return WebFoundTextRange {
+        WebFoundTextRange::DOMData {
+            range.location,
+            range.length },
+        frame->pathToFrame(),
+        0 // the order is set by the UI process
+    };
+}
 
-    HashMap<WebCore::FrameIdentifier, Vector<WebFoundTextRange>> frameMatches;
-    for (auto& simpleRange : findMatches) {
-        Ref document = simpleRange.startContainer().document();
+static inline bool canConvertToWebFoundTextRange(SimpleRange& range)
+{
+    Ref document = range.startContainer().document();
 
-        RefPtr element = document->documentElement();
-        if (!element)
-            continue;
+    RefPtr element = document->documentElement();
+    if (!element)
+        return false;
 
-        RefPtr frame = document->frame();
-        if (!frame)
-            continue;
+    RefPtr frame = document->frame();
+    if (!frame)
+        return false;
 
-        // FIXME: We should get the character ranges at the same time as the SimpleRanges to avoid additional traversals.
-        auto range = characterRange(makeBoundaryPointBeforeNodeContents(*element), simpleRange, WebCore::findIteratorOptions());
-        auto domData = WebFoundTextRange::DOMData { range.location, range.length };
-        // order set by UI process
-        auto foundTextRange = WebFoundTextRange { domData, frame->pathToFrame(), 0 };
-        m_cachedFoundRanges.add(foundTextRange, simpleRange.makeWeakSimpleRange());
+    return true;
+}
 
-        const auto frameID = frame->frameID();
-        auto& matches = frameMatches.ensure(frameID, [] {
-            return Vector<WebFoundTextRange> { };
-        }).iterator->value;
+#if ENABLE(PDF_PLUGIN)
+static inline Vector<WebFoundTextRange::PDFData> findPdfMatchesInFrame(Frame* frame, const String& string, const OptionSet<FindOptions>& options)
```

Comment:
PDF is an acronym, `OptionSet<FindOptions>` can be passed by value since it fits in a register.

```suggestion
static inline Vector<WebFoundTextRange::PDFData> findPDFMatchesInFrame(Frame* frame, const String& string, OptionSet<FindOptions> options)
```